### PR TITLE
Add iteration functionality

### PR DESCRIPTION
**Pull request by gaurav-arya**
Adds functionality for iterating through a functor. See the docstring for more info!

### CODE REVIEW

**Comment by gaurav-arya** [comment]
Any obstacles to merging this?

There are probably QOL improvements (`fiter`?) + generalizations (this could probably be a special case of a `ReduceWalk`?), but I think this feature should be easy to backcompat with any future improvements.

File: src/walks.jl [hunk 1]
```diff
@@ -166,3 +166,61 @@ function (walk::CollectWalk)(recurse, x)
...
+```@jldoctest iterate
+julia> y = ([8, 7, 6], 5, (4, 3, [2, 1]));
+
+julia> make_zipped_iterator(x, y) = zip(make_iterator(x), make_iterator(y));
+
+julia> zipped_iter = fmap(make_zipped_iterator, x, y; walk=Functors.IterateWalk(), cache=nothing);
+
+julia> collect(zipped_iter)
+8-element Vector{Tuple{Int64, Int64}}:
+8-element Vector{Tuple{Int64, Int64}}:
```
Comment (CarloLucibello) [inline src/walks.jl:209]:
```suggestion
```

**Comment by CarloLucibello** [comment]
This seems ok to add. Tests are missing.

**Review by CarloLucibello — APPROVED** [review]

**Comment by CarloLucibello** [comment]
thanks!

**Pull request merged**